### PR TITLE
I didn't really understand NSProxy

### PR DESCRIPTION
**Pull request by garrettmoon**
But I think I do now :)

Previously I inherited from NSObject because I didn't think I needed
anything fancy, but it turns out NSProxy is *less* fancy than NSObject
and allows for overriding more methods. This change is probably purely
academic in that we don't use this class to forward NSObject specific
messages, but I believe this is the 'right thing'.

### CODE REVIEW

**Comment by keleftheriou** [comment]
Build finished.

File: AsyncDisplayKit/Details/ASWeakProxy.m [hunk 1]
```diff
@@ -32,4 +32,24 @@ - (id)forwardingTargetForSelector:(SEL)aSelector
...
+  // returns nil, a selector not found exception is raised. The string argument to -signatureWithObjCTypes: outlines
+  // the return type and arguments to the message. To return a dud NSMethodSignature, pretty much any signature will
+  // suffice. Since the -forwardInvocation call will do nothing if the delegate does not respond to the selector,
+  // the dud NSMethodSignature simply gets us around the exception.
+  return methodSignature ?: [NSMethodSignature signatureWithObjCTypes:"@^v^c"];
+}
+
+- (void)forwardInvocation:(NSInvocation *)invocation
+{
+  // If we are down here this means _interceptor and _target where nil. Just don't do anything to prevent a crash
```
Comment (maicki) [inline AsyncDisplayKit/Details/ASWeakProxy.m]: Can you remove _interceptor from the comment.

**Comment by maicki** [comment]
One small comment - otherwise LGTM.

**Comment by Adlai-Holler** [comment]
This is my favorite diff

**Comment by Adlai-Holler** [comment]
Failure is unrelated, merging!

**Pull request merged**